main.js

```diff
@@ -102,26 +101,22 @@ function enableLayoutChangeEvents() {
 		};
 
 		// Generate media queries for each
-		breakpoints.forEach((width, size) => {
-			switch(size) {
-				case 'S':
-					setupQuery(`(min-width: ${ width }) and (max-width: ${ decr1(breakpoints.get('M')) })`, size);
-					break;
-				case 'M':
-					setupQuery(`(min-width: ${ width }) and (max-width: ${ decr1(breakpoints.get('L')) })`, size);
-					break;
-				case 'L':
-					setupQuery(`(min-width: ${ width }) and (max-width: ${ decr1(breakpoints.get('XL')) })`, size);
-					break;
-				case 'XL':
-					setupQuery(`(min-width: ${ width })`, size);
-					break;
-				case 'default':
-				default:
-					setupQuery(`(max-width: ${ decr1(breakpoints.get('S')) })`, size);
-					break;
+		const decr1 = val => `${Number(val.replace('px', '') - 1)}px`;
+		for (let index = 0; index < breakpoints.length; index++) {
+			const breakpoint = breakpoints[index];
```

Comment:
This is a suggestion, it doesn't block merging the pull-request --
I find it easier to read if we name the variables in the breakpoint array instead of referring to them by their index i.e. `layoutName` instead of `breakpoint[0]`.
I can't make a github code suggestion which changes this for the whole loop for some reason, github errors saying suggestions don't work on deleted lines of code

```suggestion
			const [layoutName, layoutWidth] = breakpoints[index];
```

Reply:
good suggestion